### PR TITLE
Ahnfikd7/create accept reject endpoint

### LINKED ISSUE #22: Create accept/reject applicant endpoint
### Background

When reviewers review an app and make a decision to either move the applicant to the next round or reject them for the current cycle, there should be an endpoint that they can call in order to do this.

### Task

Create a new endpoint which updates the `stage` and `step` of the applicant's application depending on whether the applicant should move on to the next round or is rejected (the `ApplicationStatus` should be updated to contain the correct values as shown in the screenshot below).

Essentially, the application's `status` should update as follows:

Image

### Request Schema

`POST /apps/decision`

```json
{
  "applicantId": number,
  "decision": "ACCEPT" | "REJECT"
}
```

### Response Schema

`204 No Content`

If the given decision was applied.

`400 Bad Request`

If any of the following are true:
* The given request body does not pass validation (`applicantId` is not a number, etc.)
* A user with the given `applicantId` does not exist

`401 Unauthorized`

If any of the following are true:
* If the calling user is not a recruiter or admin

### PR DESCRIPTION
**Pull request by ahnfikd7**
### ℹ️ Issue

Closes #22 

### 📝 Description

Wrote a new function processDecision inside applications.service that updates the application stage of an applicant to the next stage or to rejected.
In the decision endpoint inside the applications.controller, it delegates the decision making to service i.e processDecision.

### CODE REVIEW

File: apps/backend/src/applications/applications.controller.ts [hunk 1]
```diff
@@ -38,6 +39,36 @@ export class ApplicationsController {
...
+    @Body('decision') decision: 'ACCEPT' | 'REJECT',
+    @Request() req,
+  ): Promise<void> {
+    //Authorization check for admin and recruiters
+    if (![UserStatus.ADMIN, !UserStatus.RECRUITER].includes(req.user.status)) {
+      throw new UnauthorizedException();
+    }
+
+    //Check if the string decision matches with the Decision enum
+    const decisionEnum: Decision = Decision[decision as keyof typeof Decision];
```
Comment (chromium-52) [inline apps/backend/src/applications/applications.controller.ts:55]: ❓We definitely need to make sure that the `decision` we get is a valid `Decision` but I _think_ line 46 in this file (where we type the `decision` parameter) tells Nest to check this for us and throw an error if the type doesn't match. Can you try commenting out this check and passing in an invalid value as the `decision` and see if the endpoint still responds with a `400 Bad Request` error?

**Review by chromium-52 — COMMENTED** [review]
💯 Just a couple comments but looks good to me overall!

File: apps/backend/src/app.module.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { ApplicationsModule } from './applications/applications.module';
   imports: [
     TypeOrmModule.forRoot({
       type: 'postgres',
+      database: 'c4c-ops',
```
[the remark this answers is not in the document]
Comment (kimharr24) [reply apps/backend/src/app.module.ts:16]: Agreed, this change breaks for us.

File: apps/backend/src/applications/applications.controller.ts (hunk 1 above)
Comment (kimharr24) [reply apps/backend/src/applications/applications.controller.ts:55]: ⛏️A potentially better alternative altogether is to make an application decision dto whose fields are `applicantId` and `decision.` Then, you can add decorators from `class-validator` such as `@IsEnum(Decision)` to the fields to ensure that Nest performs type checks at runtime.

**Review by kimharr24 — COMMENTED** [review]
Nice work! Just a few suggestions.

**Review by chromium-52 — APPROVED** [review]

**Pull request merged**